src/string_util.cpp

```diff
@@ -283,6 +284,59 @@ string get_date_iso8601(time_t tm)
   return buf;
 }
 
+bool get_unixtime_from_iso8601(const char* pdate, time_t& unixtime)
+{
+  if(!pdate){
+    return false;
+  }
+
+  struct tm tm;
+  char*     prest = strptime(pdate, "%Y-%m-%dT%T", &tm);
+  if(prest == pdate){
+    // wrong format
+    return false;
+  }
+  unixtime = mktime(&tm);
+  return true;
+}
+
+bool convert_unixtime_from_option_arg(const char* argv, time_t& unixtime)
+{
+  if(!argv){
+    return false;
+  }
+  unixtime = 0;
+  const char* ptmp;
+  for(ptmp = argv; ptmp && *ptmp; ++ptmp){
+    if('0' <= *ptmp && *ptmp <= '9'){
+      unixtime *= 10;
+      unixtime += static_cast<time_t>(*ptmp - '0');
+    }else{
+      if('Y' == *ptmp){
+        unixtime *= (60 * 60 * 24 * 365);  // average 365 day / year
+      }else if('M' == *ptmp){
+        unixtime *= (60 * 60 * 24 * 30);   // average 30 day / month
+      }else if('D' == *ptmp){
+        unixtime *= (60 * 60 * 24);
+      }else if('h' == *ptmp){
+        unixtime *= (60 * 60);
+      }else if('m' == *ptmp){
+        unixtime *= 60;
+      }else{
+        return false;
+      }
+      if('\0' != *(ptmp + sizeof(char))){
+        return false;
+      }
+      break;
```

Comment:
This ignores more complex strings, e.g., 12h30m, or any junk data, 123jkl.  Maybe it should error instead?

Reply:
I was wondering if only simple parameters are enough, but considering the possibility of future corrections, I will try to make it complicated.